### DISCUSSION TITLE
Adjusting the area to the proportions of the icon

### DISCUSSION THREAD

**Discussion by ZigglerPL** [community/community#104904] [Programming Help]
### Select Topic Area

Question

### Body

Hello, what should be the proportions of the icon path in relation to the visible area defined in the SVG element? I want to achieve an effect where the entire area around the icon's path is adjusted to it. Just like in the case of a container that would have its width set to auto (See Twitter example
![twitter iconn](https://github.com/community/community/assets/124935764/ea6e210a-c3c6-4b4e-9d42-f220cda30cb2)
). I have a problem that the area around the path extends beyond the SVG element and I don't know why.
`<div class="icon-container">
   <svg viewBox="0 0 20 20">
       <g>
         <path d="M 1 0 L 19 9.5 L 1 19">
       </g>
   </svg>
</div> 

.icon-container {
 display: flex;
 justify-content: center;
 align-items: center;
 width: 20px;
 height: 20px;
}

SVG {
 display: inline-block;
 stroke: rgb(141, 145, 145);
 stroke-width: 1.4px;
 height: 100%;
 width: 100%;
 fill: none;
}`

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @ZigglerPL ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the GitHub product Discussions.

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Reply by ZigglerPL** [reply]
Ok, thanks.

**Comment by DanielWren48** [comment]
Hi @ZigglerPL,
Your container may have some margin or padding styles that are being applied thus the svg is not the full width/height.

You can trying adding the following to your css file:

```
* {
    margin: 0;
    padding: 0;
}
```

and
```
.icon-container, SVG {
    box-sizing: border-box;
}
```

**Reply by ZigglerPL** [reply]
The problem is that the SVG element fits the size of the container, but the size of the path element does not fit the size of the d attribute, but the path size extends beyond the area of ​​the SVG element.

**Comment by DanielWren48** [comment]
Do you set the width and height inside viewBox on the svg?
```
<svg viewBox="0 0 width height" xmlns="" class="">
    <!-- SVG -->
</svg>
```

**Reply by ZigglerPL** [reply]
Yes, you can look at all the code I included in the question.

**Comment by Praticia** [comment]
Xc5TmR7YszWHuy4HJ9r7TvY2GFuk5dwPt3
